pac4j-core/src/test/java/org/pac4j/core/profile/service/InMemoryProfileServiceTests.java

```diff
@@ -0,0 +1,138 @@
+package org.pac4j.core.profile.service;
+
+import org.apache.shiro.authc.credential.DefaultPasswordService;
+import org.junit.*;
+import org.pac4j.core.exception.*;
+import org.pac4j.core.profile.CommonProfile;
+import org.pac4j.core.util.TestsConstants;
+import org.pac4j.core.credentials.UsernamePasswordCredentials;
+import org.pac4j.core.credentials.password.PasswordEncoder;
+import org.pac4j.core.credentials.password.ShiroPasswordEncoder;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import static org.junit.Assert.*;
+
+/**
+ * Tests the {@link InMemoryProfileService}.
+ *
+ * @author Elie Roux
+ * @since 2.0.0
+ */
+public final class InMemoryProfileServiceTests implements TestsConstants {
+
+	private static final String TEST_ID = "testId";
+	private static final String TEST_LINKED_ID = "testLinkedId";
+	private static final String TEST_USER = "testUser";
+	private static final String TEST_USER2 = "testUser2";
+	private static final String TEST_PASS = "testPass";
+	private static final String TEST_PASS2 = "testPass2";
+	private static final String IDPERSON1 = "idperson1";
+	private static final String IDPERSON2 = "idperson2";
+	private static final String IDPERSON3 = "idperson3";
+
+	public final static PasswordEncoder PASSWORD_ENCODER = new ShiroPasswordEncoder(new DefaultPasswordService());
+	public final static InMemoryProfileService<CommonProfile> inMemoryProfileService = new InMemoryProfileService<CommonProfile>(x -> new CommonProfile());
+	static {
+		inMemoryProfileService.setPasswordEncoder(PASSWORD_ENCODER);
+	}
+
+
+	@BeforeClass
+	public static void setUp() {
```

Comment:
use a simple `@Before` and a non-static `InMemoryProfileService` so that you are 100% sure there is no interferences between tests :)